### PR TITLE
Fix UI issues in template listing pages

### PR DESCRIPTION
**Pull request by brionmario**
## Purpose
1. Expert mode has to be separated to a different section in IDP template listing.
2. Services representation is not clear in IDP card component.
3. Template related components are duplicated in multiple places.

## Goals
This PR fixes the above mentioned UI issues and introduces a new `TemplateGrid` component to be reused across IDP and Application template listing views.

## Approach

**IDP template listing**

Screen Shot 2020-04-24 at 6 01 29 AM

**Application template listing**

Screen Shot 2020-04-24 at 2 13 48 PM

### CODE REVIEW

File: modules/react-components/src/grid/template-grid.tsx [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        paginationLimit,
+        paginationOptions,
+        subHeading,
+        tagsSectionTitle,
+        templates,
+        templateIcons,
+        type
+    } = props;
+
+    const [ templateList, setTemplateList ] = useState<any>([]);
```
Comment (NipuniBhagya) [inline modules/react-components/src/grid/template-grid.tsx:124]: Shall we add a proper type for the state variable?

File: modules/react-components/src/card/template-card.tsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+     */
+    name: string;
+    /**
+     * Template ID.
+     */
+    id?: string;
+    /**
+     * Template image.
+     */
+    image?: any;
```
Comment (NipuniBhagya) [inline modules/react-components/src/card/template-card.tsx:63]: shall we type this properly?

File: modules/i18n/src/translations/ta-IN/portals/common.ts [hunk 3]
```diff
@@ -83,6 +83,7 @@ export const common: CommonNS = {
     revokeAll: "அனைத்தையும் நீக்கு",
     samples: "மாதிரிகள்",
     save: "சேமி",
+    services: "சேவைகள்",
```
Comment (brionmario) [inline modules/i18n/src/translations/ta-IN/portals/common.ts:86]: @thivi please verify

File: modules/i18n/src/translations/ta-IN/portals/common.ts [hunk 4]
```diff
@@ -93,6 +94,7 @@ export const common: CommonNS = {
     startsWith: "தொடங்குவது",
     submit: "சமர்ப்பி",
     switch: "மாற்று",
+    technologies: "தொழில்நுட்பங்கள்",
```
Comment (brionmario) [inline modules/i18n/src/translations/ta-IN/portals/common.ts:97]: @thivi please verify

File: modules/react-components/src/grid/template-grid.tsx [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+     */
+    subHeading?: ReactNode;
+    /**
+     * Title for the tags section.
+     */
+    tagsSectionTitle?: TemplateCardPropsInterface["tagsSectionTitle"];
+    /**
+     * List of templates.
+     */
+    templates: any;
```
Comment (NipuniBhagya) [inline modules/react-components/src/grid/template-grid.tsx:66]: Let's type this properly. Please check other places as well.

File: modules/react-components/src/grid/template-grid.tsx (hunk 1 above)
Comment (brionmario) [reply modules/react-components/src/grid/template-grid.tsx]: Fixed with commit `2ab4dc8`

File: modules/react-components/src/grid/template-grid.tsx (hunk 5 above)
Comment (brionmario) [reply modules/react-components/src/grid/template-grid.tsx]: Fixed with commit `2ab4dc8`

File: modules/react-components/src/card/template-card.tsx (hunk 2 above)
Comment (brionmario) [reply modules/react-components/src/card/template-card.tsx]: Fixed with commit `3b53328`

File: modules/i18n/src/translations/ta-IN/portals/common.ts (hunk 4 above)
Comment (thivi) [reply modules/i18n/src/translations/ta-IN/portals/common.ts:97]: startsWith: "இவற்றுடன் தொடங்கும்"

File: modules/i18n/src/translations/ta-IN/portals/common.ts (hunk 3 above)
Comment (thivi) [reply modules/i18n/src/translations/ta-IN/portals/common.ts:86]: This looks good.

Comment (brionmario) [reply modules/i18n/src/translations/ta-IN/portals/common.ts:86]: Thanks

File: modules/i18n/src/translations/ta-IN/portals/common.ts (hunk 4 above)
Comment (brionmario) [reply modules/i18n/src/translations/ta-IN/portals/common.ts:97]: Let's fix this in a later PR.

**Pull request merged**